js/client-battle.js

```diff
@@ -50,6 +50,7 @@
 			this.$chat = this.$chatFrame.find('.inner');
 
 			this.$options = this.battle.scene.$options.html('<div style="padding-top: 3px; padding-right: 3px; text-align: right"><button class="icon button" name="openBattleOptions" title="Options">Battle Options</button></div>');
+			this.game = new BattleGame(this);
```

Comment:
It's concerning to me that you're just adding this to every single battle, rather than only YouTube rooms.

Anyway I don't think `|queryresponse|` and creating a new object is necessarily the right approach. I think probably a `|controlyoutube|` protocol message type, which is handled by the battle event handler is the right way to do it.

Reply:
> It's concerning to me that you're just adding this to every single battle, rather than only YouTube rooms.

Yes, my intent was to make this a spot where plugins that use GameRoom could share code. (I also only attach YouTube code for YouTube rooms by way of the queryresponse).

> Anyway I don't think |queryresponse| and creating a new object is necessarily the right approach. I think probably a |controlyoutube| protocol message type, which is handled by the battle event handler is the right way to do it.

If you prefer that, I can do it. To me, it seems uglier to let plugin code bleed into battle code, though.